### PR TITLE
LPS-141976 / LPS-107367 / LPS-141143 Apply friendly URL tag to Blogs and Journal

### PR DESCRIPTION
**Pull request by adolfopa**

### CODE REVIEW

**Review by boton — COMMENTED** [review]
@adolfopa is ready to go with you! 
Well, it's already with you, I made a push directly to your branch :D 

- [x] The history button doesn't look good in Blogs and Journal.
- [x] The history modal doesn't open in Journal (there are some dependencies missing)

The modal is working for on layout admin web (Journal) without any changes and I do not discover any missing dependencies 🤔 

The last commit is only to test the current state on blogs 899c95e.

**Comment by boton** [comment]
I add js dependencies in `package.json` fb70256 and update again your remote branch

**Comment by adolfopa** [comment]
@boton I made additional changes so that the history works for Blogs and Layout, but there are a couple of frontend issues that we need to work on.

In Journal, the history button is misaligned. To see it working, revert the last commit.
journal-history

In blogs, after restoring an old friendly URL, we need to update the contents of the friendly URL field to reflect that (as is done in layout).

Let me know if you need more information.

**Comment by boton** [comment]
@adolfopa now we can restore non-localizable inputs like blogs. 

The position of the button in a small sidebar doesn't look easy and isn't implemented yet. 
We can rethink this in the future adding more size options and positioning the button in a better way.

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by boton** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:reevaluate:1357401_215

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by boton** [comment]
```
     [exec] * What went wrong:
     [exec] Execution failed for task ':apps:friendly-url:friendly-url-taglib:packageRunTest'.
     [exec] > Process 'command '/opt/dev/projects/github/liferay-portal/build/node/bin/node'' finished with non-zero exit value 1
```

It looks like a real error, I will take a look as soon as I can.

**Comment by boton** [comment]
ci:test:relevant

**Comment by boton** [comment]
@adolfopa I update the tests, now we only test about the localized case in the frontend tests.

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
should be ci fluke^

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
fluke again.

**Comment by Hanlf** [comment]
ci:test:relevant

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/util/comparator/FriendlyURLEntryLocalizationComparator.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
  * details.
  */
 
-package com.liferay.friendly.url.web.internal.util.comparator;
+package com.liferay.layout.admin.web.internal.util.comparator;
```
Comment (AliciaGarciaGarcia) [inline modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/util/comparator/FriendlyURLEntryLocalizationComparator.java]: you are making this comparator to layout only, but extending . why?

but letting other kind of objects, removing : 
```
!Objects.equals(_getClassName(httpServletRequest),
					Layout.class.getName())
```
					
get the friendly URL... I do not understand

File: modules/apps/friendly-url/friendly-url-api/src/main/java/com/liferay/friendly/url/util/comparator/FriendlyURLEntryLocalizationComparator.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
  * details.
  */
 
-package com.liferay.layout.admin.web.internal.util.comparator;
+package com.liferay.friendly.url.util.comparator;
```
Comment (AliciaGarciaGarcia) [inline modules/apps/friendly-url/friendly-url-api/src/main/java/com/liferay/friendly/url/util/comparator/FriendlyURLEntryLocalizationComparator.java:15]: Ok, forget my previous comment...

**Review by AliciaGarciaGarcia — COMMENTED** [review]
LGTM

**Comment by boton** [comment]
I add this task https://issues.liferay.com/browse/LPS-141143
I updated, cleaned the tests, and added a snapshot so if it fails in the future at least we will have some clue.

**Review by boton — APPROVED** [review]

**Comment by boton** [comment]
ci:test:sf

**Comment by boton** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:forward

**Comment by Hanlf** [comment]
ci:forward

**Pull request closed without merging**